contrib/labstack/echo/echotrace.go

```diff
@@ -41,11 +41,13 @@ func Middleware(opts ...Option) echo.MiddlewareFunc {
 
 			// serve the request to the next middleware
 			err := next(c)
-
-			span.SetTag(ext.HTTPCode, strconv.Itoa(c.Response().Status))
 			if err != nil {
 				span.SetTag(ext.Error, err)
+				// invokes the registered HTTP error handler
+				c.Error(err)
```

Comment:
@cgilmour After testing these changes with @jrdnull we detected an undesirable behaviour when not clearing the error after calling `c.Error(err)`. If the HTTP header is already written when running the middlewares, and something goes wrong (e.g. error committing the transaction) the default Echo error handler isn't appropriate. For that reason, some custom implementations of the HTTP error handler always handle the error, even if the response is already committed. Causing this error propagating to duplicate the action.

Looking to other middlewares, like the [logger](https://github.com/labstack/echo/blob/master/middleware/logger.go), it calls `c.Error(err)` but it doesn't propagate the error. This could be a good solution if there are no other middlewares in the chain executing after, that would do something special (e.g. transaction rollback) if there's an error. So I would suggest having an option to the developer decide if the error should be propagated or not. But I'm not sure if it's an elegant solution.

Reply:
@rafaeljusto Would it help to only call `c.Error(err)` if the response hasn't been committed yet? eg:
```go
    if !c.Response().Committed {
        c.Error(err)
    }
```
That might give the best overall behavior.

I still think it's appropriate to propagate the error so that anything further up the chain can react correctly to it.
If a developer wants to filter it, they would be able to inject their own middleware instead of it being an option in this one.

Reply:
@cgilmour Yep, adding an extra middleware to filter the error should solve the issue. Thanks! 👍